src/metadata_validator.py

```diff
@@ -346,8 +339,9 @@ def validate_required_props(self, data_record, msg_prefix):
                 if data_value is None or not str(data_value).strip():
                     result[ERRORS].append(create_error("M003",[msg_prefix, data_key], data_key, data_value))
                 else:
+                    entity_type = self.model.get_entity_type(node_type)
                     # validate program name and study name required in CRDCDH-2431.  Both are required properties.
-                    if node_type == "program" and data_key == "program_name":
+                    if entity_type == "Program" and data_key == "program_name":
```

Comment:
I assume data_key holds property name here? If so, we can not assume it is "program_name", each data commons can have their own property name. We need to add the property name in content.json file in out model repo, and use that.